packages/react-examples/src/react-components/Migrations/utils.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { Source } from '@storybook/addon-docs/blocks';
+import { makeStyles } from '@fluentui/react-make-styles';
+
+const codeComparisonUseStyles = makeStyles({
+  root: {
+    display: 'flex',
+    justifyContent: 'space-between',
+
+    '> *': {
+      width: '100%',
+    },
+
+    '> :not(:last-child)': {
+      marginRight: '5%',
+    },
+  },
+});
+
+export const CodeComparison = (props: { children: React.ReactElement[] }) => {
+  const { children } = props;
+  const classes = codeComparisonUseStyles();
+
+  return <div className={classes.root}>{children}</div>;
+};
+
+export const CodeLanguages: { [key: string]: string } = {
+  html: 'HTML',
+  css: 'CSS',
+  js: 'JavaScript',
+  jsx: 'React',
+};
+export const CodeExample = (props: { title?: string; children: React.ReactElement }) => {
+  const { title, children } = props;
+  // Access the raw values from the markdown source code block
+  const rawValue: string = children?.props?.children?.props?.children;
+  const isMarkdownCodeBlock = rawValue !== undefined;
+
+  if (isMarkdownCodeBlock) {
```

Comment:
nit: might be nicer to invert the condition so that there's less logic indented

```suggestion
  if (!isMarkdownCodeBlock) {
    return children;
```